src/server/api/routers/map_mod_publisher/mod.ts

```diff
@@ -477,6 +507,52 @@ const getGamebananaModInfo = async (gamebananaModID: number): Promise<Gamebanana
 
 
 
+const getModSearchDatabaseInfoForMod = async (gamebananaModID: number): Promise<ModSearchDatabase_ModInfo_NoId_Partial> => {
+    const modSearchDatabase = await getCurrentModSearchDatabase();
+
+
+    let modInfo: ModSearchDatabase_ModInfo_NoId_Partial | undefined = undefined;
+
+    for (const mod of modSearchDatabase) {
+        if (mod.GameBananaId === gamebananaModID) {
+            modInfo = {
+                Screenshots: mod.Screenshots,
+                Files: mod.Files,
+            };
+
+            break;
+        }
+    }
+
+
+    return modInfo ?? {};
+};
+
+
+const getModSearchDatabaseInfoForAllMods = async (mods: TrimmedMod[]): Promise<TrimmedMod[]> => {
+    const modSearchDatabase = await getCurrentModSearchDatabase();
+
+
+    outerLoop: for (const mod of mods) {
+        for (const modSearchDatabaseMod of modSearchDatabase) {
+            if (mod.gamebananaModId === modSearchDatabaseMod.GameBananaId) {
+                mod.Screenshots = modSearchDatabaseMod.Screenshots;
+                mod.Files = modSearchDatabaseMod.Files;
+                continue outerLoop;
+            }
+        }
+
+        // if the mod is not found in the mod search database, log a warning
+        console.warn(`Mod with gamebananaModId ${mod.gamebananaModId} not found in mod search database`);
+    }
+
+
+    return mods;
```

Comment:
You both modify and return mods. Is this intentional?

Reply:
Yeah. I'm modifying the objects within `mods` and then I want to return them. As I'm always returning every item in `mods`, whether or not I've modified a particular object, this seemed simpler.